pkg/netpol/connlist/conns_formatter.go

```diff
@@ -67,3 +69,57 @@ func (j jsonFormatter) writeOutput(conns []Peer2PeerConnection) (string, error)
 	}
 	return string(jsonConns), nil
 }
+
+// dotFormatter: implements the connsFormatter interface for dot output format
+type dotFormatter struct {
+}
+
+// formats an edge line from a singleConnFields struct , to be used for dot graph
+func getEdgeLine(c singleConnFields) string {
+	return fmt.Sprintf("\t%q -> %q [label=%q color=\"gold2\" fontcolor=\"darkgreen\"]\n", c.Src, c.Dst, c.ConnString)
+}
+
+// formats a peer line for dot graph
+func getPeerLine(peer eval.Peer) string {
+	var peerColor string
+	if peer.IsPeerIPType() {
+		peerColor = "red2"
+	} else {
+		peerColor = "blue"
+	}
+	peerName := peer.String()
+	return fmt.Sprintf("\t%q [label=%q color=%q fontcolor=%q]\n", peerName, peerName, peerColor, peerColor)
+}
+
+// returns a dot string form of connections from list of Peer2PeerConnection objects
+func (d dotFormatter) writeOutput(conns []Peer2PeerConnection) (string, error) {
+	edgeLines := make([]string, len(conns))      // list of edges lines
+	peersVisited := make(map[string]struct{}, 0) // acts as a set
+	var visited = struct{}{}
+	peerLines := make([]string, 0) // list of peers lines
+	for index := range conns {
+		connLine := formSingleConn(conns[index])
+		edgeLines[index] = getEdgeLine(connLine)
+		if _, ok := peersVisited[connLine.Src]; !ok {
+			peersVisited[connLine.Src] = visited
+			peerLines = append(peerLines, getPeerLine(conns[index].Src()))
+		}
+		if _, ok := peersVisited[connLine.Dst]; !ok {
+			peersVisited[connLine.Dst] = visited
+			peerLines = append(peerLines, getPeerLine(conns[index].Dst()))
+		}
+	}
+	// sort graph lines
+	sort.Strings(peerLines)
+	sort.Strings(edgeLines)
+
+	res := "digraph {\n"
+	for _, peerLine := range peerLines {
+		res += peerLine
+	}
+	for _, edgeLine := range edgeLines {
+		res += edgeLine
+	}
+	res += "}\n"
```

Comment:
be consistent with `txtFormatter.writeOutput()` : use `strings.Join` instead of concatenation with `+` where possible, and use `fmt.Sprintln("")` to get newline char (separate to a function to avoid duplication).